packages/web-components/src/components/leadspace/leadspace-body.ts

```diff
@@ -0,0 +1,197 @@
+ */
+export enum LEADSPACE_THEME {
+  /**
+   * Carbon White theme (default)
+   */
+  WHITE = 'white',
+
+  /**
+   *  Carbon Gray 100 theme
+   */
+  G100 = 'g100',
+}
+
+/**
+ * LeadSpace body.
+ */
+abstract class DDSLeadSpaceBase extends LitElement {
+  /**
+   *  Leadspace type (small, left, or centered)
+   */
+  @property({ reflect: true })
+  type = LEADSPACE_TYPE.LEFT;
+
+  /**
+   * Carbon color theme of the Leadspace
+   */
+  @property({ reflect: true })
+  theme = LEADSPACE_THEME;
+
+  /**
+   * `true` to hide the divider.
+   */
+  @property({ type: Boolean, reflect: true })
+  gradient = true;
+
+  /**
+   * Renders Leadspace copy/description
+   */
+  protected _renderCopy() {
+    const { copy } = this;
+    return html`
+      <div class="${prefix}--leadspace__row">
+        <p data-autoid="${ddsPrefix}--leadspace__desc" class="${prefix}--leadspace__desc">
+          <slot name="copy">${copy}</slot>
+        </p>
+      </div>
+    `;
+  }
+
+  /**
+   * Renders Leadspace title slot
+   */
+  protected _renderTitle() {
+    const { title } = this;
+    return html`
+      <slot name="title">${title}</slot>
+    `;
+  }
+
+  /**
+   *  Renders the image slot or the mobile image for centered leadspace
+   */
+  protected _renderImage() {
+    const { defaultSrc, alt, type } = this;
+    if (type === LEADSPACE_TYPE.CENTERED) {
+      return html`
+        <div data-autoid="${ddsPrefix}--leadspace--centered--mobile__image" class="${prefix}--leadspace--centered--mobile__image">
+          <img src="${defaultSrc}" alt="${alt}" />
+        </div>
+      `;
+    }
+    return html`
+      <slot name="image"></slot>
+    `;
+  }
+
+  /**
+   * The alternate text.
+   */
+  @property()
+  alt = '';
+
+  /**
+   * The image source.
+   */
+  @property({ attribute: 'default-src' })
+  defaultSrc = '';
+
+  /**
+   * The leadspace copy.
+   */
+  @property()
+  copy = '';
+
+  /**
+   * The leadspace title.
+   */
+  @property()
+  title = '';
+
+  render() {
+    return html`
+      <div style="${this._getBackgroundImage()}" class="${prefix}--leadspace__container ${this.getTypeClass()}">
+        <div class="${prefix}--leadspace__overlay ${this._getGradientClass()}">
+          <div class="${prefix}--leadspace--content__container">
+            <div class="${prefix}--leadspace__row">
+              <h1 class="${prefix}--leadspace__title">${this._renderTitle()}</h1>
+            </div>
+            <div class="${prefix}--leadspace__content">
+              ${this._renderCopy()}
+              <slot name="buttons"></slot>
+            </div>
+          </div>
+        </div>
+        ${this._renderImage()}
+      </div>
+    `;
+  }
+
+  /**
+   * Returns a class-name based on the gradient parameter type
+   *
+   * @protected
+   */
+  protected _getGradientClass() {
+    return this.gradient ? `${prefix}--leadspace--gradient` : '';
+  }
+
+  /**
+   * Returns a class-name based on the type parameter type
+   *
+   * @protected
```

Comment:
We add `@protected` annotation only if we cannot do the annotation via TypeScript:

```suggestion
```